api/lib/infrastructure/serializers/xml/xml-organization-learner-set.js

```diff
@@ -72,7 +74,7 @@ function _mapStudentInformationToOrganizationLearner(studentNode) {
     middleName: _getValueFromParsedElement(studentNode.PRENOM2),
     thirdName: _getValueFromParsedElement(studentNode.PRENOM3),
     sex: _convertSexCode(studentNode.CODE_SEXE),
-    birthdate: moment(studentNode.DATE_NAISS, 'DD/MM/YYYY').format('YYYY-MM-DD') || null,
+    birthdate: dayjs(studentNode.DATE_NAISS?.[0], 'DD/MM/YYYY').format('YYYY-MM-DD') || null,
```

Comment:
C'est un peu mystérieux ce qui se passe ici pour moi.

Reply:
Je t'avoue que ça été un peu mystérieux pour moi aussi.
Ce que j'ai constaté est que le parser XML ne renvoie pas une valeur mais un tableau de valeur.
Il est possible que cette entrée n'existe pas dans le fichier xml d'entrée, aussi il faut se prémunir d'un tableau non défini.
`momentjs` accepte un tableau en entrée et parse la première case de celui-ci.
Pour le vérifier, entrer cette ligne dans la console sur la page de documentation de [moment](https://momentjs.com/)
```
>> moment(['14/12/2023'], 'DD/MM/YYYY').format('YYYY-MM-DD')
<- "2023-12-14" 
```

Ce n'est pas le cas de `dayjs` qui retourne `Invalid Date` si le paramètre d'entrée est un tableau.
Là aussi, on peut l'expérimenter avec la console de la page de documentation](https://day.js.org/).
```
>> dayjs(['14/12/2023'], 'DD/MM/YYYY').format('YYYY-MM-DD')
<- "Invalid Date" 
```

Je prend toute suggestion qui permettrait de clarifier le code.

Reply:
Tu as du t'amuser avec celle là 😅
Pour la suggestion, je dirais juste une variable intermédiaire, pour qu'on comprenne à minima qu'on ne récupère pas une partie de la date de naissance, mais bien la date de naissance entière.
```suggestion
const incomingBirthDate = studentNode.DATE_NAISS?.[0];
    birthdate: dayjs(incomingBirthDate, 'DD/MM/YYYY').format('YYYY-MM-DD') || null,
```